scripts/hid_configurator/configurator_core.py

```diff
@@ -384,87 +476,37 @@ def dfu_sync_wait(dev, args, is_active):
                 time.sleep(DFU_SYNC_INTERVAL)
             else:
                 break
-    return dfu_info
 
+    return dfu_info
 
-def perform_dfu(dev, args):
-    dfu_image = args.dfu_image
-    event_id = (EVENT_GROUP_DFU << GROUP_FIELD_POS) | (DFU_DATA << TYPE_FIELD_POS)
-    recipient = get_device_pid(args.device_type)
-
-    if not os.path.isfile(dfu_image):
-        print('DFU image file does not exists')
-        return
 
+def dfu_transfer(dev, recipient, dfu_image, progress_callback):
     img_length = os.stat(dfu_image).st_size
-    if img_length <= 0:
-        print('DFU image is empty')
-        return
+    dfu_info = dfu_sync_wait(dev, recipient, False)
 
-    print('DFU image size: {} bytes'.format(img_length))
-
-    # Check there is no other DFU operation.
-    dfu_info = dfu_sync_wait(dev, args, False)
-    if dfu_info is None:
-        print('Cannot start DFU, device not responding')
-        return
+    if not is_ok(dfu_image):
+        return False
 
-    if dfu_info[0] != 0:
-        print('Cannot start DFU, already in progress')
-        return
+    if is_other_dfu_operation(dfu_info):
+        return False
 
-    # Check if the previously interrupted DFU operation can be resumed.
+    offset = find_offset(dfu_image, dfu_info)
```

Comment:
crc is calculated twice - you could pass img_csum to find_offset

Reply:
done